### PR TITLE
Ssal grant application

### PR DESCRIPTION
**Pull request by MatteoPerona**
#### Project Abstract

Ssal introduces a blockchain-based commodities exchange platform, designed to operate within the Polkadot ecosystem using smart contracts. Ssal aims to enhance transparency and trust in commodity trading while reducing entry barriers for producers and consumers in emerging markets. It leverages blockchain technology to provide real-time price transparency, decentralized price discovery, improved risk management, enhanced liquidity, and quality assurance through partnerships with local businesses. The project's primary objective is to bring equitable and efficient financial services to underserved economies, particularly in West Africa.

The development roadmap can be broken into three main sections each requiring one month of work: development, testing, and implementation. Through the first two months our team will be working out of San Diego to develop and test our application, while the last month will be spent implementing, refining, and documenting our process. Future plans involve enhancing infrastructure for continuous integration, adding customizable Automated Market Makers (AMMs), and implementing a governance system for dispute resolution. The project aligns with the Web3 Foundation's interest in funding projects that expand the use case for ink Smart Contracts and aims to benefit both its user base and the broader ecosystem by open-sourcing its code and fostering decentralized commerce development. The team has diverse expertise in blockchain, fintech, mathematics, and computer science, and we have previously worked together on projects for social good through our non-profit organization, COMPASS.

Overall, Ssal represents a promising initiative to transform commodity trading by harnessing the capabilities of blockchain technology and reducing market inefficiencies while addressing the unique needs of emerging markets.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@matteo-p:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by MatteoPerona** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md) before we can accept your contribution. Please submit the following text as a separate comment:
> 
> I have read and hereby sign the Contributor License Agreement.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

I have read and hereby sign the Contributor License Agreement.

**Comment by MatteoPerona** [comment]
recheck

**Comment by MatteoPerona** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by MatteoPerona** [comment]
Recheck

**Comment by MatteoPerona** [comment]
recheck

**Review by Noc2 — CHANGES_REQUESTED** [review]
(changes requested in the private room)

File: applications/ssal-commods-dex.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+We built the original version of this project, Riso, on Substrate, but realized that architecting a fully fledged parachain goes out of scope for the purposes of our application. Our team is looking for a grant to build Ssal using smart contracts running on Substrate-based chains. We believe that the technology is ready to be applied to a practical, real-world application. We intend to use the grant money to travel on-site in Ivory Coast this winter and set up a trial exchange between a local farmer and distributor whom we have been in contact with. In short, we would like to provide the ecosystem with a user-facing, commercial product, and hopefully, by documenting our process, we can inspire others to do the same.
```
Comment (semuelle) [inline applications/ssal-commods-dex.md:15]: The grants program doesn't fund traveling. Please see our [FAQ](https://grants.web3.foundation/docs/faq#what-activitiespositions-do-you-fund) and [application guidelines](https://grants.web3.foundation/docs/Introduction/intro) for details on what we fund.

File: applications/ssal-commods-dex.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+- **FTE:**  2,3
+- **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | GPLv3 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** in our web documentation that explains how a user can interact with our smart contracts through CLI. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Smart Contracts | We will write the requisite ink! smart contracts to create, buy, and sell commodities contracts, testing them on Shibuya testnet. |
```
Comment (semuelle) [inline applications/ssal-commods-dex.md:197]: Could you add more details about the design of the contracts? What data is required on-chain, which functions they will expose, how they interact, etc.

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @MatteoPerona. There are some things missing and not fitting, so I left a couple of comments below. Generally, I'd be curious to know if you are planning to deploy on Astar then, or possibly Watr?

File: applications/ssal-commods-dex.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+6. Designating which party (the buyer or the seller) is responsible for transportation and its related costs.
+
+This open model allows for markets which better serve the niche requirements of commodity producers and purchasers. It creates an upgradeable set of conventions which allow the market to optimize itself.
+The lockup model adds a layer of security for buyers. It freezes the buyer's funds until a contract’s finality date, at which point, the buyer will either verify that the physical product exists, is of good quality, and is changing hands, or they deny the quality/existence of the physical product and their funds are returned. For all transactions, complete or incomplete, there is an on-chain record of the transaction, so either party can use such information to take legal action against the other if they feel compelled to do so. A blockchain provides the ideal paper-trail to build a solid case and would disincentivize fraudulent activity.
```
[the remark this answers is not in the document]
Comment (MatteoPerona) [reply applications/ssal-commods-dex.md:75]: No, I will revise the wording in the document. We plan to implement a governance system to resolve disputes, but only after we have finished developing and proving the core technology.

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Requested changes in private chat room.

**Review by semuelle — CHANGES_REQUESTED** [review]
I left a comment in the chat.

**Comment by keeganquigley** [comment]
Thanks for the adjustments @MatteoPerona I will mark as ready for review and ping the rest of the committee.

**Review by semuelle — APPROVED** [review]

**Review by keeganquigley — DISMISSED** [review]
Thanks for the changes @MatteoPerona LGTM.

**Review by keeganquigley — APPROVED** [review]

**Review by semuelle — APPROVED** [review]

**Pull request merged**